lib/usecases/store_auth_token_use_case.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:flutter_survey/model/login_model.dart';
+import 'package:flutter_survey/usecases/base/base_use_case.dart';
+import 'package:flutter_survey/database/secure_storage.dart';
+import 'package:injectable/injectable.dart';
+
+@Injectable()
+class StoreAuthTokenUseCase extends DataSaverUseCase<LoginModel> {
+  final SecureStorage _secureStorage;
+
+  const StoreAuthTokenUseCase(
+    this._secureStorage,
+  );
+
+  @override
+  void save(LoginModel data) async {
+    try {
+      await _secureStorage.writeSecureData(accessTokenKey, data.accessToken);
+      await _secureStorage.writeSecureData(refreshTokenKey, data.refreshToken);
+      await _secureStorage.writeSecureData(tokenTypeKey, data.tokenType);
+    } catch (exception) {
+      throw Failed(UseCaseException(exception));
+    }
+  }
+}
```

Comment:
Having a new `DataSaverUseCase` from the base use case still does not look good to me. How about making this use case extends `UseCase<void, LoginModel>` instead?

```suggestion
class StoreAuthTokenUseCase extends UseCase<void, LoginModel> {
  final SecureStorage _secureStorage;

  const StoreAuthTokenUseCase(
    this._secureStorage,
  );

  @override
  Future<Result<void>> call(LoginModel input) async {
    try {
      await _secureStorage.writeSecureData(accessTokenKey, input.accessToken);
      await _secureStorage.writeSecureData(refreshTokenKey, input.refreshToken);
      await _secureStorage.writeSecureData(tokenTypeKey, input.tokenType);
      return Success(null);
    } catch (exception) {
      return Failed(UseCaseException(exception));
    }
  }
}
```